libs/designsystem/src/lib/components/segmented-control/segmented-control.component.ts

```diff
@@ -150,4 +151,25 @@ export class SegmentedControlComponent<TItem extends SegmentItem = SegmentItem>
       });
     }
   }
+
+  focusNativeButton(event: UIEvent) {
+    (event.target as HTMLIonSegmentButtonElement)?.setFocus();
+  }
+
+  getTabIndex(item: TItem, index: number) {
+    // When focused prevent tab stop from inner native button to outer ion-segment-button:
+    if (this.ionSegmentElement.nativeElement.matches(':focus-within')) return -1;
+    // Allow tab stop on selected item:
+    if (item.id === this.value?.id) return null;
+    // Allow tab stop on first item if no value is set:
+    if (!this.value && index === 0) return null;
+    // Prevent tab stop on all other items:
+    return -1;
+  }
+
+  @HostListener('focusin', ['$event'])
+  @HostListener('focusout', ['$event'])
+  _noop() {
```

Comment:
Consider making this explicitly private so it does not show up in the component type
```suggestion
  private noop() {
```

Reply:
Actually a really good point - but:

1. We don't use that pattern anywhere else - instead we prefix with `_` to denote as private/internal
2. Marking the method as private results in a `_noop is declared but its value is never read.` ts warning

As I recall there used to be a time where Angular insisted on the method being public 🤔  It doesn't seem that's a requirement anymore, but maybe that's why all methods decorated with `@HostListener` in our codebase are public? 🤷🏻

Reply:
For me the warning is acceptable in this case, but I see your point about  consistency!